lib/beautiful/colorful.lua

```diff
@@ -0,0 +1,1395 @@
+        self.lightness,
+        self.alpha
+    })
+end
+
+--- Return the Hue, Saturation, and Lightness
+-- @treturn number Hue [0,360]
+-- @treturn number Saturation [0,1]
+-- @treturn number Lightness [0,1]
+-- @usage local h,s,l = c.as:hsl()
+function color.as:hsl()
+    return unpack({
+        self.hue,
+        self.saturation,
+        self.lightness,
+    })
+end
+
+--- Return the Red, Green, Blue, and Alpha
+-- @treturn number Red [0,1]
+-- @treturn number Green [0,1]
+-- @treturn number Blue [0,1]
+-- @treturn number Alpha [0,1]
+-- @usage local r,g,b,a = c.as:rgba()
+function color.as:rgba()
+    return unpack({
+        self.red,
+        self.green,
+        self.blue,
+        self.alpha
+    })
+end
+
+--- Return the Red, Green, and Blue
+-- @treturn number Red [0,1]
+-- @treturn number Green [0,1]
+-- @treturn number Blue [0,1]
+-- @usage local r,g,b = c.as:rgb()
+function color.as:rgb()
+    return unpack({
+        self.red,
+        self.green,
+        self.blue,
+    })
+end
+
+-- }}} Multi-return output
+
+-- @section end
+
+-- }}} Color output functions
+
+--- Color metamethods
+-- @section color_meta
+
+color_mt.__index = color
+color_mt.__tostring = color.as.hex
+
+--- Shorthand invert of a color
+-- @function color.__unm
+-- @see beautiful.colorful.color.invert
+color_mt.__unm = color.invert
+
+--- Add the RGBA values of two colors
+-- @function color.__add
+-- @tparam lhs beautiful.colorful.color A color object
+-- @tparam rhs beautiful.colorful.color A color object
+-- @treturn beautiful.colorful.color A new color object with RGBA values added
+-- @usage colorful.color("#123") + colorful.color("#321") -- returns #444444
+color_mt.__add = function(lhs, rhs)
+    if type(rhs) ~= "table" then return nil end
+    return color.from.rgba(
+        (lhs.red + rhs.red) * 255,
+        (lhs.green + rhs.green) * 255,
+        (lhs.blue + rhs.blue) * 255,
+        (lhs.alpha + rhs.alpha) * 255
+    )
+end
+
+--- Subtract the RGBA values of two colors
+-- @function color.__sub
+-- @tparam lhs beautiful.colorful.color A color object
+-- @tparam rhs beautiful.colorful.color A color object
+-- @treturn beautiful.colorful.color A new color object with RGBA values
+-- subtracted
+-- @usage colorful.color("#123") - colorful.color("#321") -- returns #000022
+color_mt.__sub = function(lhs, rhs)
+    if type(rhs) ~= "table" then return nil end
+    return color.from.rgba(
+        (lhs.red - rhs.red) * 255,
+        (lhs.green - rhs.green) * 255,
+        (lhs.blue - rhs.blue) * 255,
+        (lhs.alpha - rhs.alpha) * 255
+    )
+end
+
+--- Compare the lightness value of two colors
+-- @function color.__lt
+-- @tparam lhs beautiful.colorful.color A color object
+-- @tparam rhs beautiful.colorful.color A color object
+-- @treturn boolean True if lhs lightness is less than rhs lightness
+-- @usage colorful.color("#ffcc99") < colorful.color("#aabbcc") -- returns false (0.8 < 0.73333)
+color_mt.__lt = function(lhs, rhs)
```

Comment:
Using the lightness for comparsion seems arbitrary... Why the lightness? People can just compare `.lightness` directly, if they want.

Reply:
It was just a thought. I figured comparing colors with < and > would be comparing lightness. Can be removed.